app/components/Activity/ActivityActions/ActivityRefresh.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect, useRef } from 'react'
+import PropTypes from 'prop-types'
+import { injectIntl } from 'react-intl'
+import styled from 'styled-components'
+import delay from 'lib/utils/delay'
+import { Button, Spinner } from 'components/UI'
+import Sync from 'components/Icon/Sync'
+import messages from './messages'
+
+const StyledButton = styled(Button)`
+  color: ${props => (props.active ? props.theme.colors.lightningOrange : null)};
+  &:hover {
+    color: ${props => props.theme.colors.lightningOrange};
+  }
+`
+StyledButton.propTypes = {
+  active: PropTypes.bool
+}
+
+const ActivityRefresh = injectIntl(({ intl, onClick, ...rest }) => {
+  const [status, setStatus] = useState(null)
+  const buttonRef = useRef()
+
+  async function fetchActivity() {
+    await onClick()
+    await delay(1000)
+    setStatus('done')
+  }
+
+  useEffect(() => {
+    if (status === 'fetching') {
+      fetchActivity()
+    }
+  }, [status])
```

Comment:
Do you think it will be cleaner if combined into one `useEffect` since it basically analyzes the same `status` flag and decided what to do based on it's value?

Reply:
I think in the tip you are referring to they recommend to split `unrelated` code. In this case I'd say it's a regular action in redux terms: start fetching, fetch, some UI feedback when done. So usually a monolithic piece of code. I don't think it's critical by any means,  so up to you.

Reply:
I've updated this as per your suggestion. It does read better when combined into a single hook in this case.